### PR TITLE
Add support for deleting EBS snapshots that back an AMI

### LINKED ISSUE #105: Add support for deleting an AMI along with its backing snapshot
I've been using Terratest in anger recently and have spotted a few things about the library.

As part of building a pipeline for some Packer/Terraform infrastructure, it's common that AMIs will be created that are instantly destined for the bin - either because a branch/PR is being tested or because the AMI failed testing. It'd be useful to add a method that is capable of both deregistering the AMI as well as deleting the snapshot behind the AMI.

### PR DESCRIPTION
**Pull request by wjam**
Fixes #105

### CODE REVIEW

**Comment by wjam** [comment]
Output from test: https://gist.github.com/wjam/ef7339ba190d950310350f5b59ebff88

File: modules/aws/ami.go [hunk 1]
```diff
@@ -17,6 +18,44 @@ const (
...
+	}
+	return snapshots
+}
+
+// GetEbsSnapshotsForAmi retrieves the EBS snapshots which back the given AMI
+func GetEbsSnapshotsForAmiE(t *testing.T, region string, ami string) ([]string, error) {
+	logger.Logf(t, "Retrieving EBS snapshots backing AMI %s", ami)
+	ec2Client, err := NewEc2ClientE(t, region)
+	if err != nil {
+		return []string{}, err
```
Comment (brikis98) [inline modules/aws/ami.go]: Can simplify to `return nil, err`. Applies to the code below too.

File: test/packer_basic_example_test.go [hunk 2]
```diff
@@ -12,7 +12,7 @@ func TestPackerBasicExample(t *testing.T) {
 	t.Parallel()
 
 	// Pick a random AWS region to test in. This helps ensure your code works in all regions.
-	awsRegion := aws.GetRandomRegion(t, nil, nil)
+	awsRegion := "eu-central-1" //aws.GetRandomRegion(t, nil, nil)
```
Comment (brikis98) [inline test/packer_basic_example_test.go]: I'm guessing this was just for testing?

If so, you can override the behavior of `aws.GetRandomRegion` without changing any code by using the `TERRATEST_REGION` env var.

File: test/packer_basic_example_test.go [hunk 3]
```diff
@@ -28,5 +28,13 @@ func TestPackerBasicExample(t *testing.T) {
 	amiID := packer.BuildAmi(t, packerOptions)
 
 	// Clean up the AMI after we're done
-	defer aws.DeleteAmi(t, awsRegion, amiID)
+	defer deleteAmi(t, awsRegion, amiID)
+}
+
+func deleteAmi(t *testing.T, awsRegion string, amiID string) {
```
Comment (brikis98) [inline test/packer_basic_example_test.go]: Perhaps we should add a `DeleteAmiAndAllSnapshots` method to the Terratest API that does exactly this?

File: test/packer_basic_example_test.go (hunk 2 above)
Comment (wjam) [reply test/packer_basic_example_test.go]: Yes it was. Only a few regions in the account I'm using has a default VPC

File: modules/aws/ami.go [hunk 4]
```diff
@@ -17,6 +18,74 @@ const (
...
+
+// GetEbsSnapshotsForAmi retrieves the EBS snapshots which back the given AMI
+func GetEbsSnapshotsForAmiE(t *testing.T, region string, ami string) ([]string, error) {
+	logger.Logf(t, "Retrieving EBS snapshots backing AMI %s", ami)
+	ec2Client, err := NewEc2ClientE(t, region)
+	if err != nil {
+		return nil, err
+	}
+
+	images, err := ec2Client.DescribeImages(&ec2.DescribeImagesInput{
```
[the remark this answers is not in the document]
Comment (wjam) [reply modules/aws/ami.go]: It doesn't look like the DescribeImages is paginated (CLI documentation says it is but the Go SDK doesn't have it as a paginated version of the method), but we're only expecting one image to be returned as we are passing in a specific AMI id

[the remark this answers is not in the document]
Comment (brikis98) [reply modules/aws/ami.go]: Understood, thx!

**Comment by brikis98** [comment]
https://github.com/gruntwork-io/terratest/releases/tag/v0.9.9

**Pull request merged**